src/renderers/dom/shared/devtools/ReactDOMUnknownComponentDevtool.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactDOMUnknownComponentDevtool
+ */
+
+'use strict';
+
+var ReactComponentTreeDevtool = require('ReactComponentTreeDevtool');
+var warning = require('warning');
+var ReactDOMFactories = require('ReactDOMFactories');
+var didWarnAboutCase = false;
+
+function handleElement(debugID, element) {
+  if (element == null || typeof element.type !== 'string' ||
+    element.type.match(/\-|\s+/) !== null ||
+    ReactDOMFactories.hasOwnProperty(element.type)) {
```

Comment:
@Aweary Agreed, as far as I know there's none.
`ReactDOMFactories` was used since it's already being used by the project.